tests/test_tag_parser.py

```diff
@@ -2414,3 +2496,361 @@ def test_spread_list_literal_as_attribute(self):
                 "...[val1]",
             ],
         )
+
+    def test_dynamic_expressions(self):
```

Comment:
There's a lot of trivial code changes in this file as the  `TagValueStruct` class now accepts a `Parser`, to be able to evaluate template tags and filters.

But the genuinely new tests start here.